src/mono/wasm/runtime/workers/README.md

```diff
@@ -11,6 +11,17 @@ file (or `.js`) and a `xyz/` subdirectory with additional sources.
 To add a new web worker, add a definition here and modify the
 `rollup.config.js` to add a new configuration.
 
+## Caveats: a note about pthreads
+
+The workers in this directory are completely standalone from the Emscripten pthreads! they do not have access to the shared instance memory, and do not load the Emscripten `dotnet.js`.  As a result, the workers in this directory cannot use any of pthread APIs or otherwise interact with the runtime in any way, except through message passing, or by having something in the runtime set up their own shared array buffer (which would be inaccessible from wasm).
+
+On the other hand, the workers in this directory also do not depend on a .NET runtime compiled with `-s USE_PTHREADS` and are thus usable on sufficiently new browser using the single-threaded builds of .NET for WebAssembly.
+
+For workers that need to interact with native code, follow the model of `../pthreads/` or `../diagnostic_server/`:
+
+- create `xyz/shared/`, `xyz/browser/` and `xyz/worker/` directories that have `index.ts` and `tsconfig.json` files that are set up for common ES, ES with DOM APIs and ES with WebWorker APIs, respectively
+- call the apropriate functions (browser or worker) from the C code or from JS.
+
```

Comment:
Could you please describe how do you enable the mocks and how to use them for testing ? Is there unit test which uses it ?

Reply:
Just to update the status: I'm working on a couple of changes to the mock stuff:
1. Move the actual mock protocol script into the sample project.  The plan is is that the .csproj `diagnostic_options.server_url` json can be `mock:relative_url/mock.js` and then we will `import(mock_url).then(module => mock(module.default).run()` - ie: the mock script is per-project and it's just an es6 module with a default export in a format that the mock engine understands.
2. I'm adding some utilities to the mock engine so it can say things like `connection.expectOkReply ((payload) => /*do a check of the payload */)` rather than leaving the mock script to parse the bytes on its own.
3. document how to use it.

Should be done by EOD today.

Reply:
ok, mocking changes are up

Reply:
I also generate `diagnostics-mock.d.ts` into the `wasm/runtime` directory in debug builds, same as `dotnet.d.ts`. I found it helpful to have TypeScript check my JS mocking script.